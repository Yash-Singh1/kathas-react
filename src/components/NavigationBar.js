import React, { Component } from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import icon from '../assets/favicon.ico';

class NavigationBar extends Component {
  render() {
    return (
      <>
        <Navbar bg='primary' variant='dark'>
          <Nav>
            <Navbar.Brand
              style={{
                fontSize: '3rem'
              }}
              href='/'
            >
              <img
                alt='kathas icon'
                src={icon}
                style={{
                  display: 'inline-block',
                  width: '2rem',
                  marginRight: 5,
                  marginLeft: 10
                }}
              />{' '}
              Kathas
            </Navbar.Brand>
            <Nav.Link href='/'>Home</Nav.Link>
            <Link className='nav-link' to='/about'>About</Link>
          </Nav>
        </Navbar>
        <div
          style={{
            marginTop: 70
          }}
        >
          {this.props.children}
        </div>
      </>
    );
  }
}

export default NavigationBar;
