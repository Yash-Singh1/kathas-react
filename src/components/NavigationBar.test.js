import React from 'react';
import NavigationBar from './NavigationBar';
import { shallow } from 'enzyme';

describe('App', () => {
  it('renders properly', () => {
    const navbar = shallow(<NavigationBar />);
    expect(navbar.instance()).toMatchSnapshot();
  });

  it('gives back children', () => {
    const navbar = shallow(
      <NavigationBar>
        <div></div>
      </NavigationBar>
    );
    expect(navbar.find('div').exists()).toBe(true);
  });
});
