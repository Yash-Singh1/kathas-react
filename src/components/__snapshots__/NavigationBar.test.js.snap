// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders properly 1`] = `"<nav style=\\"z-index: 10;\\" class=\\"navbar navbar-expand-lg navbar-dark bg-primary\\"><button aria-controls=\\"responsive-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"navbar-collapse collapse\\" id=\\"responsive-navbar-nav\\"><div class=\\"mr-auto navbar-nav\\"><form style=\\"display: flex; justify-content: center;\\" class=\\"\\"><input placeholder=\\"Search 🔍\\" aria-label=\\"Search\\" style=\\"font-size: 16px; height: 40px; max-width: 200px; margin-left: 10px; margin-right: 2px;\\" type=\\"search\\" class=\\"form-control\\" value=\\"\\"><button style=\\"width: 75px;\\" type=\\"button\\" class=\\"btn btn-success\\">Search</button></form></div></div></nav><div style=\\"margin-top: 70px;\\"></div>"`;
